examples/nodejs/ts/index.ts

```diff
@@ -1,33 +1,78 @@
 import { HMSSDK } from "@100mslive/server-sdk";
+import { HMSSession } from "@100mslive/server-sdk/dist/types/apis/interfaces/sessionInterfaces";
 
 const accessKey = process.env.HMS_ACCESS_KEY;
 const secret = process.env.HMS_SECRET;
 
 const sdk = new HMSSDK(accessKey, secret);
-const destinationService = sdk.getDestinationService();
-
-async function sampleHlsCode() {
-  // start hls and get m3u8 url back
-  const hlsIdentifier = "referenceId"; // use this to call stop
-  const appUrl = "https://www.youtube.com/watch?v=dQw4w9WgXcQ";
-  try {
-    const { url: urlWithStart } = await destinationService.startHLSAndGetUrl({
-      identifier: hlsIdentifier,
-      appUrl,
-    });
-    console.log(
-      `started hls for appUrl - ${appUrl} at hls url - ${urlWithStart}`
-    );
-
-    const { url: urlLater } = await destinationService.getHlsState({
-      identifier: hlsIdentifier,
-    });
-    console.log("hls url retrieved later", urlLater);
-  } catch (err) {
-    console.error("failed to start hls and get url ", err);
+
+// Get Auth Token
+sdk
+  .getAuthToken({
+    roomId: "test_room_id",
+    role: "host",
+    userId: "test_user",
+  })
+  .then((tokenObj) => console.log(`Auth Token: ${tokenObj.token}`));
+
+// Get Management Token
+sdk
+  .getManagementToken()
+  .then((tokenObj) => console.log(`Management Token: ${tokenObj.token}`));
+
+// Room Operations
+async function roomOperationsExample() {
+  // get the RoomService instance
+  const roomService = sdk.getRoomService();
+
+  // create a new room with config
+  const testRoom = await roomService.createRoom({
+    name: "test_room",
+    description: "this is a test room created from the NodeJs SDK example",
+  });
+
+  // retrieve the room by the name
+  const sameRoom = await roomService.getRoomByName("test_room");
+  if (testRoom == sameRoom) {
+    console.log("this is the room we just created");
   }
+}
 
-  await destinationService.stopHLS({ identifier: hlsIdentifier });
+// Active Room Operations
+async function activeRoomOperationsExample() {
+  // get the ActiveRoomService instance
+  const activeRoomService = sdk.getActiveRoomService();
+
+  // get the peer object from an active room
+  const activePeersRecord = await activeRoomService.getActivePeers(
+    "test_room_id"
+  );
+  console.log(activePeersRecord);
+
+  // send a message to a specific peer
+  const lastPeerId = Object.keys(activePeersRecord).at(-1);
+  await activeRoomService.sendMessage("test_room_id", {
+    message: "hello last peer!",
+    peerId: lastPeerId,
+  });
+}
+
+// Session Operations
+async function sessionOperationsExample() {
+  // get the SessionService instance
+  const sessionService = sdk.getSessionService();
+
+  // get all sessions
+  const allSessionsIterable = await sessionService.getAllSessionsIterable();
```

Comment:
let's use async iterators, the code on using side would look much simpler

```
const allSessions = sessionService.getAllSessions();
for await (const session of allSessions) {
    console.log(session);
}
```

Reply:
allSesions can have a field to denote the page size and if it has to load the next value or it's already available

```
const sessionsIterable = sessionService.getSessionsIterable();
for await (const session of sessionsIterable) {
    console.log(session);
    if (!sessionsIterable.isNextAvailable) {
       console.log("the next loop is gonna take some time");
    }
}
```